server/services/banner/service.ts

```diff
@@ -61,13 +62,15 @@ export class BannerService extends ServiceClass implements IBannerService {
 	}
 
 	async getNewBannersForUser(userId: string, platform: BannerPlatform, bannerId?: string): Promise<IBanner[]> {
-		const { roles } = await this.Users.findOneById(userId, { projection: { roles: 1 } });
+		const user = await this.Users.findOneById<Pick<IUser, 'roles'>>(userId, { projection: { roles: 1 } });
+
+		const { roles } = user || { roles: [] };
 
 		const banners = await this.Banners.findActiveByRoleOrId(roles, platform, bannerId).toArray();
 
 		const bannerIds = banners.map(({ _id }) => _id);
 
-		const result = await this.BannersDismiss.findByUserIdAndBannerId(userId, bannerIds, { projection: { bannerId: 1, _id: 0 } }).toArray();
+		const result = await this.BannersDismiss.findByUserIdAndBannerId<Pick<IBannerDismiss, '_id' | 'bannerId'>>(userId, bannerIds, { projection: { bannerId: 1, _id: 0 } }).toArray();
```

Comment:
`projection` for this one actually has `_id: 0` .. so I guess the correct `Pick` would be:
```suggestion
		const result = await this.BannersDismiss.findByUserIdAndBannerId<Pick<IBannerDismiss, 'bannerId'>>(userId, bannerIds, { projection: { bannerId: 1, _id: 0 } }).toArray();
```